lib/dpp/DriveStateRepository.js

```diff
@@ -242,34 +248,49 @@ class DriveStateRepository {
    * @return {Promise<boolean>}
    */
   async verifyInstantLock(instantLock) {
-    return instantLock.verify(this.simplifiedMasternodeList.getStore());
-
-    // TODO: Enable with feature flag:
-    // const header = this.blockExecutionContext.getHeader();
-    //
-    // let coreChainLockedHeight;
-    // if (header) {
-    //   ({ coreChainLockedHeight } = header);
-    // }
-    //
-    // try {
-    //   const { result: isVerified } = await this.coreRpcClient.verifyIsLock(
-    //     instantLock.getRequestId().toString('hex'),
-    //     instantLock.txid,
-    //     instantLock.signature,
-    //     coreChainLockedHeight,
-    //   );
-    //
-    //   return isVerified;
-    // } catch (e) {
-    //   // Invalid address or key error or
-    //   // Invalid, missing or duplicate parameter
-    //   if ([-8, -5].includes(e.code)) {
-    //     return false;
-    //   }
-    //
-    //   throw e;
-    // }
+    const header = await this.blockExecutionContext.getHeader();
+
+    if (!header) {
+      return false;
+    }
+
+    const {
+      height: blockHeight,
+    } = header;
+
+    const verifyLLMQSignaturesWithCoreFeatureFlag = await this.getLatestFeatureFlag(
+      this.featureFlagTypes.VERIFY_LLMQ_SIGS_WITH_CORE,
+      blockHeight,
+      this.getDBTransaction('documents'),
+    );
+
+    if (!verifyLLMQSignaturesWithCoreFeatureFlag || !verifyLLMQSignaturesWithCoreFeatureFlag.get('enabled')) {
+      return instantLock.verify(this.simplifiedMasternodeList.getStore());
```

Comment:
Please left some comments that we use dashcore lib to verify signatures, but this approach doesn’t handle signatures created by old quorums so that’a why we switch to Core RPC method via feature flag